applications/HDF5Application/custom_io/hdf5_connectivities_data.cpp

```diff
@@ -90,7 +90,10 @@ void ConnectivitiesData::CreateEntities(NodesContainerType& rNodes,
         }
         ElementType::Pointer p_elem =
             r_elem.Create(mIds[i], nodes, rProperties(mPropertiesIds[i]));
-        rElements.push_back(p_elem);
+        // here we can safely use the insert with the rElements.end() as the hint
+        // because, when reading, we can assume that it was written in the
+        // sorted order within HDF5.
+        rElements.insert(rElements.end(), p_elem);
```

Comment:
this is not efficient as you may be creating it out of order

Reply:
It cannot be out of order, because we assume HDF5 files are written by the `HDF5Application`, hence when we write, we always write in the sorted order.

Reply:
wait, mIds is ordered?

Reply:
i'll accept the comment, but ... what if somone writes the hdf5 "by hand"? in that case it will be enormously expensive, correct?

Reply:
Yes, it will be expensive, but on the other hand, do we expect them to create HDF5 files by their own?, it needs to have a special structure. It it not impossible, but tedious.

If they are creating, then they can create them in sorted manner as well.